src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1228,7 +1228,7 @@ protected function callScope(callable $scope, $parameters = [])
         // query as their own isolated nested where statement and avoid issues.
         $originalWhereCount = count($query->wheres);
 
-        $result = call_user_func_array($scope, $parameters) ?: $this;
+        $result = $scope(...$parameters) ?: $this;
```

Comment:
We need to be positive that parameters never has string keys... otherwise unpacking will throw an error.